### PR TITLE
Setting up project infrastructure

### PR DESCRIPTION
**Pull request by SimonbJohnson**

### CODE REVIEW

File: .config/coveragerc
Comment (turnerm) [inline .config/coveragerc:1]: This isn't needed since we don't have any tests

File: .config/pre-commit-config.yaml
Comment (turnerm) [inline .config/pre-commit-config.yaml:1]: If you run `pre-commit autoupdate` you can bring all the hooks to the latest version

File: .config/ruff.toml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+]
+# List of rules: https://beta.ruff.rs/docs/rules/
+select = [
+  "E",   # pycodestyle - default
+  "F",   # pyflakes - default
+  "I"    # isort
+]
+
+[isort]
+known-local-folder = ["hdx"]
```
Comment (turnerm) [inline .config/ruff.toml:14]: These lines can be removed

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  workflow_dispatch:
+  release:
+    types: [published]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
```
Comment (turnerm) [inline .github/workflows/publish.yaml:13]: `checkout` now has v4

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+# Versioning
+
+[tool.hatch.version]
+source = "vcs"
+
+[tool.hatch.version.raw-options]
+local_scheme = "no-local-version"
+version_scheme = "python-simplified-semver"
+
+# Tests
```
Comment (turnerm) [inline pyproject.toml:71]: You can take out this section for now

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+[project.readme]
+file = "README.md"
+content-type = "text/markdown"
+
+[project.urls]
+Homepage = "https://github.com/OCHA-DAP/hapi-schemas"
+
+[project.optional-dependencies]
+timezones = ["hdx-python-api"]
```
Comment (turnerm) [inline pyproject.toml:47]: This should be "hdx-python-database", hopefully that will trim down the requirements file a bit

**Review by turnerm — CHANGES_REQUESTED** [review]
This looks great! In addition to the changes I've specified, could you:

* run pre-commit on all the files? For example, I see that a few have a missing newline, which should be fixed by one of the hooks.  
* update the changelog

By the way were you able to test the new GHA hook? IIRC you have to push to main first before the action will appear on the menu. In that case we can just cross our fingers and merge, and if there are any bugs we'll fix them :)

**Comment by SimonbJohnson** [comment]
@turnerm - I've updated for the request changes.  I've not tested the github actions yet, as you say it is not available until in main.

File: src/hapi_schema/databasequeries.py
Comment (turnerm) [inline src/hapi_schema/databasequeries.py:1]: Oops one last thing, can you remove this file, it's not needed

**Comment by turnerm** [comment]
I realized that we forgot to properly implement the extras because there is no import check. However Mike's changes in HAPI 215 mean we won't have to have the extras anymore so I think that's fine.

**Pull request merged**